lib/datasource/local/settings_storage.dart

```diff
@@ -64,13 +70,14 @@ class _SettingsStorage {
 
   set inRecoveryMode(bool value) => _preferences.setBool(_kInRecoveryMode, value);
 
-  set _accountName(String? value) => _preferences.setString(_kAccountName, value ?? '');
-
-  set privateKey(String? value) {
-    _secureStorage.write(key: _kPrivateKey, value: value);
-    if (value != null) {
-      _privateKey = value;
-    }
+  set _accountName(String? value) {
+    _preferences.setString(_kAccountName, value ?? '');
+    // Retrieve accounts list
+    final List<String> accts = accountsList;
+    // Add new account
+    accts.add(accountName);
```

Comment:
This should make sure the account name we are setting is not already in the list, otherwise this list gets longer and longer and longer

Reply:
Done